eth/downloader/queue.go

```diff
@@ -334,7 +334,13 @@ func (q *queue) Results(block bool) []*fetchResult {
 	}
 	// Using the newly calibrated result size, figure out the new throttle limit
 	// on the result cache
-	throttleThreshold := uint64((common.StorageSize(blockCacheMemory) + q.resultSize - 1) / q.resultSize)
```

Comment:
It appears we stop early if there are no completed items (line 298).  Can you explain how `resultSize` would be 0 here?